src/main/java/com/mturko/bestshop/controller/ProductController.java

```diff
@@ -12,12 +13,20 @@
 @RequestMapping(value = "products")
 public class ProductController {
 
-    private final ProductService productService;
+    @Autowired
+    private ProductService productService;
```

Comment:
Replace field injection to constructor-based auto-wiring method. Just move annotation @Autowired to the respective constructor. See for details: https://www.baeldung.com/constructor-injection-in-spring